WordPress/Classes/Services/PostService.swift

```diff
@@ -0,0 +1,599 @@
+import Foundation
+
+class PostService : LocalCoreDataService {
+    static let ErrorDomain = "PostServiceErrorDomain"
+    static let DefaultNumberToSync: UInt = 40
+
+    enum PostType: String {
+        case any = "any"
+        case post = "post"
+        case page = "page"
+
+        func type() -> AbstractPost.Type {
+            switch self {
+            case .post: return Post.self
+            case .page: return Page.self
+            case .any: return AbstractPost.self
+            }
+        }
+    }
+
+    typealias SuccessPosts = ([AbstractPost]) -> Void
+    typealias FailureBasic = (NSError?) -> Void
+    typealias SuccessBasic = () -> Void
+
+    convenience override init() {
+        self.init(managedObjectContext: ContextManager.sharedInstance().mainContext)
+    }
+
+    func makePost(for blog: Blog) -> Post? {
+        return make(type: Post.self, blog: blog)
+    }
+
+    func makePage(for blog: Blog) -> Page? {
+        return make(type: Page.self, blog: blog)
+    }
+
+    func make(postType: PostType?, blog: Blog) -> AbstractPost? {
+        return make(type: postType?.type() ?? Post.self, blog: blog)
+    }
+
+    func make<T: AbstractPost>(type postType: T.Type, blog: Blog) -> T? {
+        assert(self.managedObjectContext == blog.managedObjectContext, "Blog's context should be the the same as the service's")
+
+        guard let post = NSEntityDescription.insertNewObjectForEntityForName(postType.entityName(), inManagedObjectContext: self.managedObjectContext) as? T else {
+            return nil
+        }
+
+        post.blog = blog
+        post.remoteStatus = AbstractPostRemoteStatusSync
+
+        guard let postPost = post as? Post else {
+            return post
+        }
+
+        postPost.postFormat = blog.settings?.defaultPostFormat
+        postPost.postType = Post.typeDefaultIdentifier
+
+        let postCategoryService = PostCategoryService(managedObjectContext:self.managedObjectContext)
+        if let category = postCategoryService.findWithBlogObjectID(blog.objectID, andCategoryID: (blog.settings?.defaultCategoryID)!) {
```

Comment:
Should handle a case in which `blog.settings` or `defaultCategoryID` might be `nil`?

I'm thinking this might be a race condition between a blog sync finishing (or error) vs this method being called along the way, causing a crash.